src/test/ui/nll/closure-requirements/escape-argument-callee.stderr

```diff
@@ -2,7 +2,7 @@ note: no external requirements
   --> $DIR/escape-argument-callee.rs:26:38
    |
 LL |         let mut closure = expect_sig(|p, y| *p = y);
-   |                                      ^^^^^^
+   |                                      ^^^^^^^^^^^^^
```

Comment:
This is odd, I would have assumed the changes to cause the opposite diff

Reply:
We replace `def_span` by `span_with_body` for closures, so span is longer except in specially addressed cases.  As this is a debugging feature, I did not bother to keep the former behaviour.

Reply:
The ui output regression is fine in single line cases, but for multi line closures it's quite bad (and that makes up like 2/3rd of the ui changes. I don't think we should merge this as is. If mitigations can be deployed at diagnostic rendering time that would be fine with me. Since it's all about nll diagnostics, it could be possible to fix the span at borrowck time.